components/applications-service/pkg/storage/postgres/hab_event.go

```diff
@@ -134,41 +134,159 @@ func (db *Postgres) IngestHealthCheckEventWithoutMetrics(event *habitat.HealthCh
 
 	// @afiune all our backend was designed for the health check to be all
 	// uppercases but habitat is actually sending case sensitive strings
-	newHealth := strings.ToUpper(event.GetResult().String())
+	eventHealth := strings.ToUpper(event.GetResult().String())
 	// TODO @afiune verify health
 
-	// If the service already exists, we just do a simple update
+	// If the service already exists, we just do an update
 	if exist {
-		return db.updateService(
+		return db.updateTables(
 			svc,
 			eventMetadata,
 			svcMetadata,
 			pkgIdent,
-			newHealth,
+			eventHealth,
 		)
 	}
 
 	// But if the service doesn't exist, we will handle it as a new service insertion
-	err1 := db.insertNewService(
+	err = db.insertNewService(
 		eventMetadata,
 		svcMetadata,
 		pkgIdent,
-		strings.ToUpper(event.GetResult().String()),
+		eventHealth,
 	)
 
-	if err1 != nil {
+	if err != nil {
 		// We will retry once if a unique constraint is hit,
 		// in case a new deployment or service group is in the same batch of events.
-		if err1.Error() == uniqueDeploymentError || err1.Error() == uniqueServiceGroupError {
+		if err.Error() == uniqueDeploymentError || err.Error() == uniqueServiceGroupError {
 			return db.insertNewService(
 				eventMetadata,
 				svcMetadata,
 				pkgIdent,
-				strings.ToUpper(event.GetResult().String()),
+				eventHealth,
 			)
 		}
 	}
-	return err1
+	return err
+}
+
+func (db *Postgres) updateTables(
+	svc *service,
+	eventMetadata *habitat.EventMetadata,
+	svcMetadata *habitat.ServiceMetadata,
+	pkgIdent *packageIdent,
+	health string) error {
+
+	db.updateService(svc, eventMetadata, svcMetadata, pkgIdent, health)
```

Comment:
let's make sure the service always gets some update so the database has a record of when the last message was received for a service

Reply:
Fixed! Thanks for the clarification @danielsdeleo - Since this is the only missing thing I am going to merge it after buildkite is green, having said that we are now always updating the service timestamp. 💯